plan_entitlements.go

```diff
@@ -0,0 +1,157 @@
+	Entitlement  *PlanEntitlement  `json:"entitlement,omitempty"`
+	Entitlements []PlanEntitlement `json:"entitlements,omitempty"`
+}
+
+func (c *Client) PlanEntitlement() *PlanEntitlementRequest {
+	return &PlanEntitlementRequest{
+		client: c,
+	}
+}
+
+func (sr *PlanEntitlementRequest) GetList(ctx context.Context, planCode string) (*PlanEntitlementResult, *Error) {
+	subPath := fmt.Sprintf("%s/%s/%s", "plans", planCode, "entitlements")
+
+	clientRequest := &ClientRequest{
+		Path:   subPath,
+		Result: &PlanEntitlementResult{},
+	}
+
+	result, clientErr := sr.client.Get(ctx, clientRequest)
+	if clientErr != nil {
+		return nil, clientErr
+	}
+
+	planEntitlementResult, ok := result.(*PlanEntitlementResult)
+	if !ok {
+		return nil, &ErrorTypeAssert
+	}
+
+	return planEntitlementResult, nil
+}
+
+func (sr *PlanEntitlementRequest) Get(ctx context.Context, planCode string, featureCode string) (*PlanEntitlement, *Error) {
+	subPath := fmt.Sprintf("%s/%s/%s/%s", "plans", planCode, "entitlements", featureCode)
+
+	clientRequest := &ClientRequest{
+		Path:   subPath,
+		Result: &PlanEntitlementResult{},
+	}
+
+	result, clientErr := sr.client.Get(ctx, clientRequest)
+	if clientErr != nil {
+		return nil, clientErr
+	}
+
+	planEntitlementResult, ok := result.(*PlanEntitlementResult)
+	if !ok {
+		return nil, &ErrorTypeAssert
+	}
+
+	return planEntitlementResult.Entitlement, nil
+}
+
+func (sr *PlanEntitlementRequest) Delete(ctx context.Context, planCode string, featureCode string) (*PlanEntitlement, *Error) {
+	subPath := fmt.Sprintf("%s/%s/%s/%s", "plans", planCode, "entitlements", featureCode)
+
+	clientRequest := &ClientRequest{
+		Path:   subPath,
+		Result: &PlanEntitlementResult{},
+	}
+
+	result, clientErr := sr.client.Delete(ctx, clientRequest)
+	if clientErr != nil {
+		return nil, clientErr
+	}
+
+	planEntitlementResult, ok := result.(*PlanEntitlementResult)
+	if !ok {
+		return nil, &ErrorTypeAssert
+	}
+
+	return planEntitlementResult.Entitlement, nil
+}
+
+func (sr *PlanEntitlementRequest) DeletePrivilege(ctx context.Context, planCode string, featureCode string, privilegeCode string) (*PlanEntitlement, *Error) {
+	subPath := fmt.Sprintf("%s/%s/%s/%s/%s/%s", "plans", planCode, "entitlements", featureCode, "privileges", privilegeCode)
+
+	clientRequest := &ClientRequest{
+		Path:   subPath,
+		Result: &PlanEntitlementResult{},
+	}
+
+	result, clientErr := sr.client.Delete(ctx, clientRequest)
+	if clientErr != nil {
+		return nil, clientErr
+	}
+
+	planEntitlementResult, ok := result.(*PlanEntitlementResult)
+	if !ok {
+		return nil, &ErrorTypeAssert
+	}
+
+	return planEntitlementResult.Entitlement, nil
+}
+
+func (sr *PlanEntitlementRequest) Replace(ctx context.Context, planCode string, input []EntitlementInput) (*PlanEntitlementResult, *Error) {
+	return sr.update(ctx, planCode, input, false)
+}
+
+func (sr *PlanEntitlementRequest) Update(ctx context.Context, planCode string, input []EntitlementInput) (*PlanEntitlementResult, *Error) {
```

Comment:
OpenAPI doc for this is quite cryptic. I believe we should provide examples in the OpenAPI to make it more understandable.

Reply:
https://github.com/getlago/lago-openapi/pull/436/files#diff-19caa0f66713124e6ea8ce21967f8dff9720cb99dfa44202d41977bd99dbb22a